playback/src/audio_backend/jackaudio.rs

```diff
@@ -71,7 +71,10 @@ impl Open for JackSink {
 
 impl Sink for JackSink {
     fn write(&mut self, packet: &AudioPacket, converter: &mut Converter) -> io::Result<()> {
-        let samples_f32: &[f32] = &converter.f64_to_f32(packet.samples());
+        let samples = packet
+            .samples()
+            .map_err(|e| io::Error::new(io::ErrorKind::Other, e.to_string()))?;
```

Comment:
Hmm, I don't know. Idiomatically, it strikes me as odd that a decoder error is cast as an I/O error. It isn't really an I/O error, is it? Thinking out loud, how about unwrapping or else returning an empty error of samples and perhaps printing an error?

Reply:
IMO it would be better to modify the `Sink` trait. 

```rust
#[derive(Error)]
struct SinkError {
    DecoderError(...),
    IoError(std::io::Error)
}
```

Reply:
Alright!